HIP/hip-792.md

```diff
@@ -0,0 +1,290 @@
+---
+hip: 792
+title: Extend isAuthorized to process current transaction in HAS System Contract
+author: Brendan Graetz <@bguiz>
+working-group: Nana Essilfie-Conduah <@nana-ec>, Luke Lee  <@lukelee-sl>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+last-call-date-time: 2023-10-22T16:00:00Z
+created: 2023-08-21
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/793
+updated: 2023-08-23
+requires: 632
+---
+
+## Abstract
+
+<!-- a short (~200 word) description of the technical issue being addressed. -->
+
+HIP-632 adds a new system contract, `hederaAccountService` that exposes a method `isAuthorized(address, messageHash, signatureBlob)`. This HIP proposes a similar version of that function, which implements the same functionality, but restricted for use *only* on the current (in flight) transaction. This exposes a new method `isAuthorizedCurrentTransaction()`.
+
+## Motivation
+
+<!-- The motivation is critical for HIPs that want to change the Hedera codebase or ecosystem. It should clearly explain why the existing specification is inadequate to address the problem that the HIP solves. HIP submissions without sufficient motivation may be rejected outright. -->
+
+`hederaAccountService.isAuthorized(address, messageHash, signatureBlob)` does work for the following scenarios:
+
+- When an account's admin key is "simple", and is comprised of either a single EdDSA key, or a single ECDSA key
+- When an account's admin key is "complex", and is comprised of multiple EdDSA keys and/or  ECDSA keys, combined using one or more `KeyList`s or `ThresholdKey`s.
+
+`hederaAccountService.isAuthorized(address, messageHash, signatureBlob)` does **not** work for the following scenarios:
+
```

Comment:
I'm not exactly clear as to why this scenario would not work with `isAuthorized` in HIP-632.  The signature blob is a protobuf serialized `SignatureMap` which I believe supports adding contract keys.

Reply:
The idea here is that the client invoking the smart contract does not need to pass in `signatureBlob`, and that any values that it needs, including `signatureBlob` are read from the current transaction.